tools/ARIAtools/ARIAProduct.py

```diff
@@ -433,12 +433,21 @@ def __mappingVersion__(self, fname, version):
                 '/science/grids/imagingGeometry/azimuthAngle'
             ]
             if version.lower()=='1c':
+                # get weather model name
+                meta = gdal.Info(fname)
+                for i in meta.split():
+                    if '/science/grids/corrections/external/troposphere/' in i:
+                        model_name = i.split('/')[-3]
+                        break
                 lyr_pref = '/science/grids/corrections'
                 sdskeys_addlyrs = [
                     lyr_pref + '/derived/ionosphere/ionosphere',
-                    lyr_pref + '/external/tides/solidEarth',
-                    lyr_pref + '/external/troposphere/troposphereWet',
-                    lyr_pref + '/external/troposphere/troposphereHydrostatic'
+                    lyr_pref + '/external/tides/solidEarth'
+                        '/reference/solidEarthTide',
+                    lyr_pref + f'/external/troposphere/{model_name}'
```

Comment:
What is the expected working behavior under scenario if the solid earth tide layer is missing but tropo is present

Reply:
So I simplified the existing check I had after I realized it didn't work as expected. Now the non-existence of such layers is captured and reported [here](https://github.com/aria-tools/ARIA-tools/blob/sss_refsec/tools/ARIAtools/ARIAProduct.py#L453) and [here](https://github.com/aria-tools/ARIA-tools/blob/sss_refsec/tools/ARIAtools/ARIAProduct.py#L477-#L486) in ARIAProduct. I confirmed it doesn't lead to issues in a test stack Brett provided which had tropo ref/sec, but no SET nor ionosphere.

Through initial sorting in ARIAProduct:
```
WARNING: Expected data layer key ionosphere not found in NETCDF:"/u/leffe-data/ssangha/solid_earth_test/temp_tropo_test/D071/products/S1-GUNW-D-R-071-tops-20200130_20200112-135156-34956N_32979N-PP-dd0c-v2_0_4.nc
WARNING: Expected data layer key solidEarthTide not found in NETCDF:"/u/leffe-data/ssangha/solid_earth_test/temp_tropo_test/D071/products/S1-GUNW-D-R-071-tops-20200130_20200112-135156-34956N_32979N-PP-dd0c-v2_0_4.nc
```

And again at TS stage:
```
WARNING: Stack for default ARIA TS layer ionosphere cannot be generated as it does not exist in any of the input products
WARNING: Stack for default ARIA TS layer solidEarthTide cannot be generated as it does not exist in any of the input products
```